### PR TITLE
Cleanup on release and release-adjacent jobs

### PR DESCRIPTION
**Pull request by justaugustus**
Some miscellaneous fixes to SIG Release/release-adjacent jobs:

- releng: Ensure Release Managers are notified on build/verify failures

  Adding the Release Managers group to alert emails for Kubernetes build
  and verify jobs. Getting alerted on these would've raised an OpenAPI
  spec update failure and build tagging issue earlier to us. We also drop
  the fork annotation from prototype-master jobs. These jobs aren't ready
  to be forked yet and checking a release branch creation run, don't seem
  to do what's intended.

- releng: Windows Testgrid tab names to explicitly reference version

  Where possible (non-generated, but forked) jobs should explicitly
  reference the versions of Kubernetes that they test against.

  We rename the Windows jobs here, drop the generic annotation
  (fork-per-release-generic-suffix: "true"), and use the correctly
  versioned images for each release branch.

- images: Symlink krte/variants.yaml to kubekins-e2e/variants.yaml

  Recently discovering images/krte/variants.yaml, I noticed that this
  variants.yaml closely mirrors the kubekins-e2e one, however it is not an
  update that the Release Managers have made in the past, and it had some
  stale entries. Instead of making checking the content between the two,
  we symlink kubekins-e2e/variants.yaml so that they're always in sync.

- config-forker: Jobs not in blocking/informing are marked as config error

  With the removal of the sig-release-x.y-all boards, any job landing on
  one of those boards should be considered a job configuration error. The
  change was already made in generate_tests.py and this is the
  accompanying change for the config-forker.

ref: https://github.com/kubernetes/kubernetes/issues/86182, https://github.com/kubernetes/test-infra/issues/15514, https://github.com/kubernetes/test-infra/pull/15612

### CODE REVIEW

**Comment by justaugustus** [comment]
Explicit hold until I figure some things out:
/hold

**Comment by justaugustus** [comment]
Ready for review:
/hold cancel
/assign @fejta @cjwagner 
cc: @kubernetes/release-engineering @kubernetes/ci-signal

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.14.yaml [hunk 1]
```diff
@@ -123,7 +125,9 @@ periodics:
       securityContext:
         privileged: true
 - annotations:
+    fork-per-release-generic-suffix: "true"
```
Comment (cpanato) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.14.yaml]: there are any documentation that describes all the annotations?

**Comment by justaugustus** [comment]
@cpanato -- Here's the doc on forked job annotations: https://github.com/kubernetes/test-infra/blob/master/experiment/config-forker/README.md

**Pull request merged**